src/sliding_sync.rs

```diff
@@ -299,9 +305,13 @@ pub fn submit_async_request(req: MatrixRequest) {
         .expect("BUG: async worker task receiver has died!");
 }
 
-
+/// Submits a request to a receiver that runs a loop to wait for successful login 
+pub enum LoginRequest{
+    LoginByPassword(LoginByPassword),
+    LoginBySSO(String)
+}
```

Comment:
minor grammar improvement
```suggestion
/// Details of a login request that get submitted within [`MatrixRequest::Login`].
pub enum LoginRequest{
    LoginByPassword(LoginByPassword),
    LoginBySSO(String)
}
```

Reply:
fixed